diff -ruw src.orig/src/m_misc.c src/src/m_misc.c
--- src.orig/src/m_misc.c	2009-10-14 18:14:40.000000000 +0200
+++ src/src/m_misc.c	2009-10-14 18:14:55.000000000 +0200
@@ -294,7 +294,7 @@
    def_int,ss_none}, // select music driver (DOS), -1 is autodetect, 0 is none"; in Linux, non-zero enables music
   {"pitched_sounds",{&pitched_sounds},{0},0,1, // killough 2/21/98
    def_bool,ss_none}, // enables variable pitch in sound effects (from id's original code)
-  {"samplerate",{&snd_samplerate},{22050},11025,48000, def_int,ss_none},
+  {"samplerate",{&snd_samplerate},{44100},11025,48000, def_int,ss_none},
   {"sfx_volume",{&snd_SfxVolume},{8},0,15, def_int,ss_none},
   {"music_volume",{&snd_MusicVolume},{8},0,15, def_int,ss_none},
   {"mus_pause_opt",{&mus_pause_opt},{2},0,2, // CPhipps - music pausing
